docs/src/opensourcetpa.md

```diff
@@ -0,0 +1,294 @@
+
+Run this command, and apparently, nothing will happen on the command line. But you will find a directory called `demo` has been created containing some files including a `config.yml` file which is a blueprint for our new deployment.
+
+## Provisioning the deployment
+
+Now we are ready to create the containers (or virtual machines) on which we will run our new deployment. This can be achieved with the `provision` command. Run:
+
+```
+tpaexec provision demo
+```
+
+You will see TPA work through the various operations needed to prepare for deployment of your configuration. 
+
+## Deploying
+
+Once provisioned, you can move on to deployment. The installs, if needed, operating systems and system packages. It then installs the requested Postgres architecture and performs all the needed configuration. 
+
+```
+tpaexec deploy demo
+```
+
+You will see TPA work through the various operations needed to deploy your configuration.
+
+
+## Testing
+
+You can quickly test your newly deployed configuration using the tpaexec `test` command which will run pgbench on your new database.
+
+```
+tpaexec test demo
+```
+
+## Connecting
+
+To get to a psql prompt, the simplest route is to log into one of the containers (or VMs or host depending on configuration) using SSH. Run 
+
+```
+tpaexec ping demo
+```
+
+to ping all the SSH connectable hosts in the deployment: You will get output that looks something like:
+
+```
+$ tpaexec ping demo 
+unfair | SUCCESS => {
+    "changed": false,
+    "ping": "pong"
+}
+uptake | SUCCESS => {
+    "changed": false,
+    "ping": "pong"
+}
+quondam | SUCCESS => {
+    "changed": false,
+    "ping": "pong"
+}
+uptight | SUCCESS => {
+    "changed": false,
+    "ping": "pong"
+}
+```
+
+Select one of the nodes which has responded with `SUCCESS`. We shall use `uptake` foir this example.
+
+Change current directory to the created configuration directory. For example, our configuration is called demo, so we go to that directory. In there, we run `ssh -F ssh_config ourhostname` to connect.
+
+```
+cd demo
+ssh -F ssh_config uptake
+Last login: Wed Sep  6 10:08:01 2023 from 172.17.0.1
+[root@uptake ~]# 
+```
+
+We can now change user to the `postgres` user using `sudo -iu postgres`. As `postgres` we can run `psql`. TPA has already configured that user with a `.pgpass` file so there's no need to present a password.
+
+```
+[root@uptake ~]# 
+postgres@uptake:~ $ psql
+psql (15.4)
+Type "help" for help.
+
+postgres=# 
+```
+
+And we are connected to our database.
+
+You can connect from the host system without SSHing into one of the containers. Obtain the IP address of the host you want to connect to from the `ssh_config` file.
+
+```
+$ grep "^ *Host" demo/ssh_config 
+Host *
+Host uptight
+    HostName 172.17.0.9
+Host unfair
+    HostName 172.17.0.4
+Host quondam
+    HostName 172.17.0.10
+Host uptake
+    HostName 172.17.0.11
+```
+
+We are gping to connect to uptake, so the IP address is 172.17.0.11. 
```

Comment:
```suggestion
We are going to connect to uptake, so the IP address is 172.17.0.11. 
```

Reply:
Ah my spell checker has died. Two things to fix.